### PR TITLE
Prevent a task from starting if an instance with the same name is run…

### LINKED ISSUE #81: Restricting Spring Cloud Task instances
There are use cases where an application is triggered by cron scheduler but due to communication with 3rd party API(s), the application runs slow and hence the cron scheduler triggers another instance of the application again while the first instance is still running.

### PR DESCRIPTION
**Pull request by cppwfs**
…ning.

feature is disabled by default.  To enable it set spring.cloud.task.singleInstanceEnabled=true.
resolves #81

### CODE REVIEW

File: spring-cloud-task-docs/src/main/asciidoc/features.adoc [hunk 1]
```diff
@@ -205,6 +205,34 @@ following property on the child task:
...
+[[features-single_instance_enabled]]
+=== Parent Task Id
+
+Spring Cloud Task allows a user to establish whether a task should execute if
+another task with the same `spring.cloud.task.name` is executing.  If
+`spring.cloud.task.single-instance-enabled` is set to true then the current
+task instance will check the datastore to see if a TaskExecution with
+the same `spring.cloud.task.name` has a task_lock associated with it.
+If so this Task instance will
+throw a TaskExecutionException prior to making any entries into the datastore.
```
Comment (mminella) [inline spring-cloud-task-docs/src/main/asciidoc/features.adoc]: Missing \` around `TaskExecutionException`

File: spring-cloud-task-samples/taskprocessor/src/test/java/io/spring/TaskProcessorApplicationTests.java [hunk 2]
```diff
@@ -49,6 +51,9 @@
 	@Autowired
 	protected MessageCollector collector;
 
+	@MockBean
+	private TaskProperties taskProperties;
+
```
Comment (mminella) [inline spring-cloud-task-samples/taskprocessor/src/test/java/io/spring/TaskProcessorApplicationTests.java]: Why is the indentation in this file all off?

File: spring-cloud-task-core/pom.xml [hunk 3]
```diff
@@ -76,6 +76,14 @@
...
 			<scope>test</scope>
 		</dependency>
-	</dependencies>
+		<dependency>
+			<groupId>org.springframework.integration</groupId>
+			<artifactId>spring-integration-core</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>org.springframework.integration</groupId>
+			<artifactId>spring-integration-jdbc</artifactId>
```
[the remark this answers is not in the document]
Comment (cppwfs) [reply spring-cloud-task-core/pom.xml]: If we use the JDBCLockRegistry it is required.

File: spring-cloud-task-core/pom.xml [hunk 4]
```diff
@@ -76,6 +76,14 @@
 			<artifactId>hibernate-validator</artifactId>
 			<scope>test</scope>
 		</dependency>
-	</dependencies>
+		<dependency>
+			<groupId>org.springframework.integration</groupId>
+			<artifactId>spring-integration-core</artifactId>
+		</dependency>
```
[the remark this answers is not in the document]
Comment (cppwfs) [reply spring-cloud-task-core/pom.xml]: Why make it optional if it is required feature of Task?

**Comment by cppwfs** [comment]
Closing invalid.

**Pull request closed without merging**